File: lib/features/production/models/production_material_usage_source.dart
class ProductionMaterialUsageSource {
  const ProductionMaterialUsageSource({
    required this.executionSegmentId,
    required this.executionSegmentCode,
    required this.shared,
    required this.canOpen,
    required this.canSettle,
    this.sourcePlanId,
  });
  final String executionSegmentId;
  final String executionSegmentCode;
  final bool shared;
  final bool canOpen;
  final bool canSettle;
  final String? sourcePlanId;

  factory ProductionMaterialUsageSource.fromJson(Map<String, dynamic> json) =>
      ProductionMaterialUsageSource(
        executionSegmentId: json['executionSegmentId'] as String,
        executionSegmentCode:
            json['executionSegmentCode'] as String? ?? '原领料任务',
        shared: json['shared'] == true,
        canOpen: json['canOpen'] == true,
        canSettle: json['canSettle'] == true,
        sourcePlanId: json['sourcePlanId'] as String?,
      );
}

File: server/src/main/java/com/uten/imp/features/stock/allocation/dto/ProductionMaterialUsageSource.java
package com.uten.imp.features.stock.allocation.dto;

import java.util.UUID;

/** Real issue-owning task, retained within its original material permissions. */
public record ProductionMaterialUsageSource(UUID executionSegmentId, String executionSegmentCode,
                                            boolean shared, boolean canOpen, boolean canSettle,
                                            UUID sourcePlanId) {
    public ProductionMaterialUsageSource(UUID executionSegmentId, String executionSegmentCode,
            boolean shared, boolean canOpen, boolean canSettle) {
        this(executionSegmentId,executionSegmentCode,shared,canOpen,canSettle,null);
    }
}
